### PR TITLE
Add Methods Section

### PR DESCRIPTION
**Pull request by samm82**
Section 4 of [my thesis](https://github.com/samm82/TestGen-Thesis/blob/methods/thesis.pdf) (with some parts being reused in Section III.D of [my paper](https://github.com/samm82/TestGen-Thesis/blob/methods/paper.pdf)) consists of discussion on how analysis is performed from the perspectives of both what categorizations are used (so far, just explaining "rigidity") and how analysis is automated/augmented with automated tools, which I based off of #82. I _think_ I did a pretty good job of explaining the underlying ideas without getting bogged down by too many details, but a review from the perspectives of people who _haven't_ written the code will put this to the test. Some notes on what is still left to do in this section as part of larger changes:

1. In addition to rigidity, I also plan to outline what the different "classes" of discrepancies mean in more depth here, which will likely introduce more categories that are split off from the "other discrepancies" category (e.g., "definitions" is likely a useful category of discrepancy to track; inspired by #86)
2. What is meant by "source categorizations" will be addressed as part of #89.
   - [ ] A note that the "rigid" graphs _also_ exclude inferred relations should be included
3. User instructions for running code will be addressed as part of #78.

This also addresses the second to-do item (plus its child items) of #69, although maybe not entirely.

### CODE REVIEW

**Comment by smiths** [comment]
@samm82 to make it completely unambiguous, can you please include a link to the pdf file that includes the section you want reviewed?  Also, please provide the section number.

**Comment by samm82** [comment]
Yup! Section 4 of [my thesis](https://github.com/samm82/TestGen-Thesis/blob/methods/thesis.pdf), with some parts being reused in Section III.D of [my paper](https://github.com/samm82/TestGen-Thesis/blob/methods/paper.pdf).

**Comment by smiths** [comment]
Feedback:

- I don't like the title of Chapter 4 (Methods).  The title doesn't tell the reader enough about the content of the chapter.  Maybe something like Graph Generation and Analysis Tool?
- The first paragraph, what I call the "intro blurb" is too short.  It is good to have a roadmap of the chapter contents, but you should say something about the purpose of the chapter.  The chapter presents the tools that was developed to facilitate the visualization and analysis of the large quantities of data collected.  The tools is critical because you will need to regenerate the graphs and update the analysis whenever there is a change to the data, either because of a correction, or because of a future change to the source documents.
- In Section 4.1 you should give an example from your source of one or two implicit discrepancies.  You point to Table 6.1, but the table has the counts of discrepancies, not examples.  It will help the reader to see what you were looking for when you read the source data and decided on an implicit discrepancy.  We want the describe your methods to the point where we could get someone else to reproduce your efforts and they would come up with essentially the same data you came up with.
- In Section 4.1 you mention the keywords you are looking for.  You should clarify where there words are.  I believe the words are in the table you created.  I believe the words were written by you when you were doing your data collection.  Another interpretation would be that the keywords are words that you searched for in the original source text.  You should aim to be unambiguous.
- In Section 4.2 the link to "our approach glossary" didn't work for me
- You should provide more detail on the entries in Table 4.1.  What does the information in brackets mean.  Sometimes it says Author and then a number, but sometimes there is just a number?  What is Name?  If it is the name of a test approach, you have room to label the column Test Approach Name.
- "All child-parent relations are graphed, as well as synonym relations where either: 1. both terms are present in the glossary".  I feel like information is missing for criteria 1.  For the both terms, are they synonyms? 
- Do you define earlier in the thesis what you mean by synonym and how they are detected?  Do you give examples.  Examples will help.
- Do you define earlier in the thesis what you mean by the child parent relation?  Do you give examples?  Examples will help.
- On page 13 you have a comment that asks "should I add an example?"  Yes.  :-)
- There is a lot going on in the first full paragraph on page 13.  You might want to split this into separate paragraphs and add detail.
- In Section 4.3 you talk about a "stricter" type.  Can you add an explanation of what you mean by this?
- Section 4.3.2, the sentence that starts with "Then, the relevant sources ..." is too long.
- Section 4.3.2, "how times" should be "how many times"
- I feel like there is a lot going on here to keep track of.  Can you create a more formal model of your graph?  In particular, can you summarize all the data you are tracking and all the possible relations between the data, and the modifiers for those relations.  I feel like all this information should be together in one spot.

I like what you are doing @samm82, but I don't think I entirely understand Chapter 4.  I think part of the problem is reading it out of context.  I don't know what you are planning on already telling the reader by the time they get to this chapter.  I also think a formal model of your graph would really help the reader, and likely the author.  Maybe @JacquesCarette will have some good ideas on how to be more formal?  We can discuss in our meeting later today.

**Comment by samm82** [comment]
Based on the workflow discussion in #92, I'm merging this PR, with the relevant content to be reviewed later
